src/workerd/jsg/jsg.h

```diff
@@ -2276,6 +2276,8 @@ class ExternalMemoryAdjustment final {
 // Isolate<TypeWrapper>::Lock. Usually this is only done in top-level code, and the Lock is
 // passed down to everyone else from there. See setup.h for details.
 
+class V8System;
```

Comment:
I don't think we should be exposing `V8System` like this. The intent of JSG is to abstract V8 specifics away as much as possible and this just exposes them more. If we have to expose something here, which I'm unsure about, then the functionality should be folded into a new `jsg::Lock` method... like `js.pumpMessageLoop()` so that the details of interfacing with the v8 APIS do not need to leak more out to other areas.

Reply:
Yeah, I agree. `js.pumpMessageLoop()` was the intended plan before I sent this out for review :sweat_smile:

Reply:
Would it be possible to make `V8System` entirely hidden here so that it's not exposed at all from jsg.h?